src/libs/actions/Task.js

```diff
@@ -0,0 +1,146 @@
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import * as Environment from '../Environment/Environment';
+
+/**
+ * Clears out the task info from the store
+ */
+function clearOutTaskInfo() {
+    Onyx.set(ONYXKEYS.TASK, null);
+}
+
+/**
+ * Assign a task to a user
+ * Function title is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} parentReportID
+ * @param {String} title
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(currentUserEmail, parentReportID, title, description, assignee) {
+    // Grab the assigneeChatReportID if there is an assignee
+    const assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]).reportID;
+
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, title, description);
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    function getEnvironmentURL() {
+        return Environment.getEnvironmentURL().then((environmentURL) => {
+            const text = `${environmentURL}/r/${optimisticTaskReport.reportID}`;
+            return text;
+        });
+    }
+
+    getEnvironmentURL().then((text) => {
+        Report.addComment(parentReportID, `[Created a task](${text}): ${title}`);
+    });
```

Comment:
This should be done with optimistic data instead of using addComment here

For example: you can use `buildOptimisticAddCommentReportAction` to build it.
You can use the same text here, but then also make sure to add the taskReportID into the message.
So like

const reportComment = ReportUtils.buildOptimisticAddCommentReportAction(text);
reportComment.reportAction.message[0].taskReportID = taskReportID